src/highdicom/pr/content.py

```diff
@@ -0,0 +1,876 @@
+                    )
+                if not hasattr(referenced_images[0], 'NumberOfFrames'):
+                    raise TypeError(
+                        'Specifying "referenced_frame_number" is not valid '
+                        'when the referenced image is not a multi-frame image.'
+                    )
+                if isinstance(referenced_frame_number, Sequence):
+                    for f in referenced_frame_number:
+                        if f < 1 or f > referenced_images[0].NumberOfFrames:
+                            raise ValueError(
+                                f'Frame number {f} is invalid for provided '
+                                'dataset.'
+                            )
+                else:
+                    f = referenced_frame_number
+                    if f < 1 or f > referenced_images[0].NumberOfFrames:
+                        raise ValueError(
+                            f'Frame number {f} is invalid for provided '
+                            'dataset.'
+                        )
+                if referenced_segment_number is not None:
+                    raise TypeError(
+                        'Specifying both "referenced_segment_number" and '
+                        '"referenced_frame_number" is not supported.'
+                    )
+            if referenced_segment_number is not None:
+                if multiple_images:
+                    raise ValueError(
+                        'Specifying "referenced_segment_number" is not '
+                        'supported with multiple referenced images.'
+                    )
+                if referenced_images[0].SOPClassUID != SegmentationStorage:
+                    raise TypeError(
+                        '"referenced_segment_number" is only valid when the '
+                        'referenced image is a segmentation image.'
+                    )
+                number_of_segments = len(referenced_images[0].SegmentSequence)
+                if isinstance(referenced_segment_number, Sequence):
+                    for s in referenced_segment_number:
+                        if s < 1 or s > number_of_segments:
+                            raise ValueError(
+                                f'Segment number {s} is invalid for provided '
+                                'dataset.'
+                            )
+                else:
+                    s = referenced_segment_number
+                    if s < 1 or s > number_of_segments:
+                        raise ValueError(
+                            f'Segment number {s} is invalid for provided '
+                            'dataset.'
+                        )
+            ref_image_seq = []
+            for im in referenced_images:
+                if (
+                    not hasattr(im, 'PixelData') and
+                    not hasattr(im, 'FloatPixelData') and
+                    not hasattr(im, 'DoubleFloatPixelData')
+                ):
```

Comment:
In 04089bc I have moved this into `highdicom.utils`. I think there are probably other similar efforts around the library that we should combine at some point

Reply:
Thanks @CPBridge! I just realized that this check is not a good idea, because in pathology we generally work only with metadata (dataset without Pixel Data element). We should use a different approach to check whether a dataset represents an image using [Image Pixel Description Macro Attributes](https://dicom.nema.org/medical/dicom/current/output/chtml/part03/sect_C.7.6.3.html#table_C.7-11c). However, we should only use the subset of attributes that are also used for images with Float Pixel Data and Double Float Pixel Data elements.

@seandoyle implemented a similar utility function in https://github.com/herrmannlab/highdicom/pull/83 (see [here](https://github.com/seandoyle/highdicom/blob/08f63a918988f5a8b90a6c26ef082f657d7a5874/src/highdicom/sr/templates.py#L627-L650)). Note that we didn't place the function in `highdicom.utils` because of circular import issues.

Reply:
Good point, now I check whether the IOD identified by the SOPClassUID contains any pixel element, rather than checking whether the specific dataset contains these attributes.